toolkit/tools/scheduler/schedulerutils/graphbuildstate.go

```diff
@@ -176,10 +208,20 @@ func (g *GraphBuildState) RecordBuildResult(res *BuildResult, allowToolchainRebu
 		g.failures = append(g.failures, res)
 	}
 
+	// 'NodeFreshnessRebuildRequired' is a special value that indicates that the node was rebuilt due to  missing files
+	// (user requested rebuilds are already at the max freshness). In this case, we want to reset the freshness to the
+	// max, so that subsequent dependant nodes will be rebuilt. Also ensure that the freshness is not greater than the max.
+	freshness := res.Freshness
+	if freshness < 0 || freshness > g.GetMaxFreshness() {
+		err = fmt.Errorf("unexpected freshness value of '%d' for node '%s'. Should be: (0 < %d <= %d)", freshness, res.Node.FriendlyName(), freshness, g.GetMaxFreshness())
```

Comment:
I'm a bit worried that `0 < 15 <= 2` (just an example) might be a bit confusing. What do you think about this:
```suggestion
		err = fmt.Errorf("unexpected freshness value of %d for node '%s'. Should be: (0 < <freshness> <= %d)", freshness, res.Node.FriendlyName(), g.GetMaxFreshness())
```